### ISSUE TITLE
Support AndroidKeyStoreECPrivateKey as Ec2SigningKey

### ISSUE THREAD

**Issue by spheroid** [android/cose-lib#21]
I haven't found a way to instantiate an `Ec2SigningKey` using an Android Keystore backend EC key because `AndroidKeyStoreECPrivateKey` does not implement `ECPrivateKey` and as you don't have access to the raw private D parameter bytes, there's no other way to construct it using the `PrivateKeyRepresentationBuilder`.

Not having the private D parameter does prevent you from serializing the key but it doesn't prevent performing cryptographic operations such as signing. Would it be possible to provide a way to instantiate the `Ec2SigningKey` without having access to the private key data – i.e. letting signature operations to succeed but serialization to fail?

**Comment by vikram-gaur** [comment]
`Ec2SigningKey` or `CoseKey` in general was designed to have cbor serialization as a core element. So it would go against that setup to have a key that can be used but not serialized.

I think at that point, it would be better to use the `AndroidKeyStoreECPrivateKey` directly rather than using `Ec2SigningKey`.

**Comment by spheroid** [comment]
Thanks for responding! I understand your point and indeed this would go against the design philosophy.

When researching other options, I looked into creating a custom implementation of the `CoseKey` that I could use with the Android Keystore backed key. But unfortunately it cannot be used with `CoseUtils.generateCoseSign1()` because it has a runtime key type check and `Ec2SigningKey` is final. An added benefit of using this helper is that it already implements the DER to COSE signature conversion logic.

I have now implemented a custom signing logic that just constructs the `Sign1Message` from scratch, using this key. In that sense I'm happy either way, but it would be nice to get rid of some of the duplicated code.
